### PR TITLE
[Fix] Move experiences to `UserSkill` from `Skill`

### LINKED ISSUE #7688: ♻️  Experiences should be accessed view UserSkill instead of Skill
## ♻️ Debt/Refactor

Experiences can be retrieved from api schema via Skill type.

## 🕵️ Details

Experiences should be retrieved via UserSkill type, and not through Skills.

`experienceSkillRecord` can stay on Skill type, as it is only non-null when skill is nested in Experience.

`experience.experienceSkillRecord` should be non-null when nested in UserSkill instead of Skill.

## 🙋‍♀️ Proposed Solution

(optional) Do you have a proposed solution?

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the bug was addressed.

- [ ] `experiences`, and each experience subtype, are removed from Skill type in api schema
- [ ] `experiences`, and each experience subtype, are added to UserSkill type in api schema
- [ ] And client-side operations which retrieved experiences through skills go through UserSkills instead
- [ ] mutations to link experiences and skills are unaffected

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #7688 

## 👋 Introduction

Moves all experience types to the `UserSkill` model instead of the `Skill` model to better reflect the relationship.

## 🏅 Bonus

We had a component that used this relationship which was not being used so it was deleted instead of being fixed.

## 🕵️ Details

There is an intersection of `User`, `Skill` and `Experience` records with the owner being the `User`. 

Previously you accessed this through deeply nested queries (`user.userSkills.skills.experiences`). This made sense since experiences are linked to a user through skills. We want to flatten this relationship (if you need to know why, ask @tristan-orourke or @esizer ).

In order to do this, we are moving the `Experience` up to the `UserSkill` type instead to get it closer to the owner (`User`). So, you can now query experiences attached to a skill via `User.userSkills.experience`.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Navigate to `/graphiql`
2. Run some queries to get `UserSkill` records
3. Confirm you can get experiences off the user skill
4. Confirm you can no longer get experiences from the `Skill` model

<details>
  <summary>Test query</summary>

```graphql
query UserSkill {
  me {
    userSkills {
      id
      experiences {
        id
      }
    }
  }
}

```

</details>

### CODE REVIEW

File: api/graphql/schema.graphql [hunk 1]
```diff
@@ -623,13 +623,6 @@ type Skill {
   keywords: SkillKeywords
   families: [SkillFamily!] @belongsToMany
 
-  # Experiences
```
[the remark this answers is not in the document]
Comment (esizer) [reply api/graphql/schema.graphql:626]: @tristan-orourke What are your thoughts about this? Do we need these relationships in the actual model anymore?

File: api/graphql/schema.graphql [hunk 2]
```diff
@@ -641,6 +634,19 @@ type UserSkill {
   skill: Skill! @belongsTo
   skillLevel: SkillLevel @rename(attribute: "skill_level")
   whenSkillUsed: WhenSkillUsed @rename(attribute: "when_skill_used")
+
+  # Experiences
+  experiences: [Experience!]
+    @with(relation: "awardExperiences")
```
Comment (tristan-orourke) [inline api/graphql/schema.graphql:640]: Nice catch on the `@with` optimization

File: packages/fake-data/src/fakeUserSkills.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+const generateUserSkill = (
+  skill: Skill,
+  experiences: AnyGeneratedExperience[],
+  uniqueEnforcerId: UniqueEnforcer,
+) => {
+  const uniqueId = uniqueEnforcerId.enforce(() => {
+    return faker.string.uuid();
+  });
+  return {
+    __typename: undefined,
```
Comment (tristan-orourke) [inline packages/fake-data/src/fakeUserSkills.ts:28]: Why not set this to the real typename?

File: packages/fake-data/src/fakeUserSkills.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    id: uniqueId,
+    skill,
+    user: mockUser,
```
Comment (tristan-orourke) [inline packages/fake-data/src/fakeUserSkills.ts:31]: I reccomend passing user in as an argument, with a default value of mockUser

**Review by tristan-orourke — APPROVED** [review]
Looks good! Just left a couple suggestions for the faker.

File: api/graphql/schema.graphql (hunk 1 above)
[the remark this answers is not in the document]
Comment (esizer) [reply api/graphql/schema.graphql:626]: Makes sense to me! Removed in [remove experience relationship from skill model](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7733/commits/d24608904ca4487da5dc1dde7b5c4a80974ec0a4)

File: packages/fake-data/src/fakeUserSkills.ts (hunk 3 above)
Comment (esizer) [reply packages/fake-data/src/fakeUserSkills.ts]: That was just a copy and paste error 😅 Good catch!

Fixed in [improve fake user skill generation](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7733/commits/16a0f8fc5c1554e77b89e2ab22f99e60d521c47c)

File: packages/fake-data/src/fakeUserSkills.ts (hunk 4 above)
Comment (esizer) [reply packages/fake-data/src/fakeUserSkills.ts]: Sounds good, I added it as an arg and set a default incase we don't care who the user is.

[improve fake user skill generation](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7733/commits/16a0f8fc5c1554e77b89e2ab22f99e60d521c47c)

**Pull request merged**